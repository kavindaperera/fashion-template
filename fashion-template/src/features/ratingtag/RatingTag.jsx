import React from "react";
import { Rating } from "semantic-ui-react";
function RatingTag({ enableRating, rating }) {
  return (
    <div data-testid="rating-tag">
      {enableRating && rating > 0 && (
        <Rating disabled defaultRating={rating} maxRating={5} />
      )}
      {enableRating && rating == 0 && (
        <p style={{ color: "grey" }}>still not rated</p>
      )}
    </div>
  );
}

export default RatingTag;
