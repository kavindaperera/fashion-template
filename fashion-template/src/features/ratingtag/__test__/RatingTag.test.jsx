import React from 'react';
import ReactDOM from 'react-dom';
import RatingTag from '../RatingTag'
import { render, cleanup } from '@testing-library/react'
import "@testing-library/jest-dom/extend-expect"
import renderer from 'react-test-renderer';
import { Rating } from "semantic-ui-react";

afterEach(cleanup);

it("ratings tag renders without crashing", ()=>{
    const div = document.createElement("div");
    ReactDOM.render(<RatingTag></RatingTag>,div)
});

it("renders rating tag correctly",()=>{
    const {getByTestId} = render(<RatingTag enableRating={true} rating={0} ></RatingTag>)
    expect(getByTestId('rating-tag')).toHaveTextContent("still not rated")
});

it("renders rating tag correctly",()=>{
    const {getByTestId} = render(<RatingTag enableRating={false} rating={0} ></RatingTag>)
    expect(getByTestId('rating-tag')).toHaveTextContent("")
});

it("renders rating tag correctly",()=>{
    const {getByTestId} = render(<RatingTag enableRating={true} rating={5} ></RatingTag>)
    expect(getByTestId('rating-tag')).toHaveTextContent("")
});

it("renders rating tag correctly",()=>{
    const {getByTestId} = render(<RatingTag enableRating={false} rating={5} ></RatingTag>)
    expect(getByTestId('rating-tag')).toHaveTextContent("")
});

it("renders rating tag correctly",()=>{
    const {getByTestId} = render(<RatingTag enableRating={true} rating={3} ></RatingTag>)
    expect(getByTestId('rating-tag')).toHaveTextContent("")
});

it("renders rating tag correctly",()=>{
    const {getByTestId} = render(<RatingTag enableRating={false} rating={3} ></RatingTag>)
    expect(getByTestId('rating-tag')).toHaveTextContent("")
});

//Snapshot Testing

it('matches snapshot 1', ()=>{
    const tree = renderer.create(<RatingTag enableRating={true} rating={0} ></RatingTag>).toJSON();
    expect(tree).toMatchSnapshot();
})

it('matches snapshot 2', ()=>{
    const tree = renderer.create(<RatingTag enableRating={false} rating={0} ></RatingTag>).toJSON();
    expect(tree).toMatchSnapshot();
})

it('matches snapshot 3', ()=>{
    const tree = renderer.create(<RatingTag enableRating={true} rating={5}></RatingTag>).toJSON();
    expect(tree).toMatchSnapshot();
})

it('matches snapshot 4', ()=>{
    const tree = renderer.create(<RatingTag enableRating={false} rating={5}></RatingTag>).toJSON();
    expect(tree).toMatchSnapshot();
})

it('matches snapshot 5', ()=>{
    const tree = renderer.create(<RatingTag enableRating={true} rating={3}></RatingTag>).toJSON();
    expect(tree).toMatchSnapshot();
})

it('matches snapshot 6', ()=>{
    const tree = renderer.create(<RatingTag enableRating={false} rating={3}></RatingTag>).toJSON();
    expect(tree).toMatchSnapshot();
})



