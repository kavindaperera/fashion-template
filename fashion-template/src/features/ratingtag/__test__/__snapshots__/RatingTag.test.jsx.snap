// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<div
  data-testid="rating-tag"
>
  <p
    style={
      Object {
        "color": "grey",
      }
    }
  >
    still not rated
  </p>
</div>
`;

exports[`matches snapshot 2 1`] = `
<div
  data-testid="rating-tag"
/>
`;

exports[`matches snapshot 3 1`] = `
<div
  data-testid="rating-tag"
>
  <div
    className="ui disabled rating"
    onMouseLeave={[Function]}
    role="radiogroup"
  >
    <i
      aria-checked={false}
      aria-posinset={1}
      aria-setsize={5}
      className="active icon"
      onClick={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      role="radio"
      tabIndex={0}
    />
    <i
      aria-checked={false}
      aria-posinset={2}
      aria-setsize={5}
      className="active icon"
      onClick={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      role="radio"
      tabIndex={0}
    />
    <i
      aria-checked={false}
      aria-posinset={3}
      aria-setsize={5}
      className="active icon"
      onClick={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      role="radio"
      tabIndex={0}
    />
    <i
      aria-checked={false}
      aria-posinset={4}
      aria-setsize={5}
      className="active icon"
      onClick={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      role="radio"
      tabIndex={0}
    />
    <i
      aria-checked={true}
      aria-posinset={5}
      aria-setsize={5}
      className="active icon"
      onClick={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      role="radio"
      tabIndex={0}
    />
  </div>
</div>
`;

exports[`matches snapshot 4 1`] = `
<div
  data-testid="rating-tag"
/>
`;

exports[`matches snapshot 5 1`] = `
<div
  data-testid="rating-tag"
>
  <div
    className="ui disabled rating"
    onMouseLeave={[Function]}
    role="radiogroup"
  >
    <i
      aria-checked={false}
      aria-posinset={1}
      aria-setsize={5}
      className="active icon"
      onClick={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      role="radio"
      tabIndex={0}
    />
    <i
      aria-checked={false}
      aria-posinset={2}
      aria-setsize={5}
      className="active icon"
      onClick={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      role="radio"
      tabIndex={0}
    />
    <i
      aria-checked={true}
      aria-posinset={3}
      aria-setsize={5}
      className="active icon"
      onClick={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      role="radio"
      tabIndex={0}
    />
    <i
      aria-checked={false}
      aria-posinset={4}
      aria-setsize={5}
      className="icon"
      onClick={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      role="radio"
      tabIndex={0}
    />
    <i
      aria-checked={false}
      aria-posinset={5}
      aria-setsize={5}
      className="icon"
      onClick={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      role="radio"
      tabIndex={0}
    />
  </div>
</div>
`;

exports[`matches snapshot 6 1`] = `
<div
  data-testid="rating-tag"
/>
`;
